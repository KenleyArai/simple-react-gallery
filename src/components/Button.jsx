import styled from "styled-components";

const Button = styled.div`
  display: block;
  width: 2rem;
  height: 2rem;
  transition: all 0.3s cubic-bezier(0.25, 0.8, 0.25, 1);
  box-shadow: 0 1px 4px 0 rgba(0, 0, 0, 0.37);

  &:hover {
    box-shadow: 0 13px 25px 0 rgba(0, 0, 0, 0.3),
      0 11px 7px 0 rgba(0, 0, 0, 0.19);
  }
`;

export default Button;
