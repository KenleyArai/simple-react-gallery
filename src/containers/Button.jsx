import React, { Component } from "react";
import PropTypes from "prop-types";

import ButtonStyle from "styled/ButtonStyle";

export default class Button extends Component {
  static propTypes = {
    clickHandler: PropTypes.func.isRequired,
    children: PropTypes.string.isRequired
  };

  render() {
    var { clickHandler, children } = { ...this.props };
    return <ButtonStyle>{children}</ButtonStyle>;
  }
}
