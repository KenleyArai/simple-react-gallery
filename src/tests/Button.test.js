import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";
import "jest-styled-components";

configure({ adapter: new Adapter() });

import Button from "containers/Button";

describe("Button", () => {
  it("should render button", () => {
    const wrapper = shallow(
      <Button clickHandler={() => {}}>Test Button</Button>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
