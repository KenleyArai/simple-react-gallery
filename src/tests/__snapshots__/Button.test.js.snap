// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render button 1`] = `
<styled.div>
  Test Button
</styled.div>
`;
